prometheus/indico_prometheus/metrics.py

```diff
@@ -0,0 +1,150 @@
+num_valid_bookings = metrics.Gauge('indico_num_valid_bookings', 'Number of valid bookings')
+num_pending_bookings = metrics.Gauge('indico_num_pending_bookings', 'Number of pending bookings')
+
+num_occurrences = metrics.Gauge('indico_num_booking_occurrences', 'Number of occurrences')
+num_valid_occurrences = metrics.Gauge('indico_num_valid_booking_occurrences', 'Number of valid occurrences')
+
+num_ongoing_occurrences = metrics.Gauge('indico_num_ongoing_booking_occurrences', 'Number of ongoing bookings')
+
+if LIVESYNC_AVAILABLE:
+    size_livesync_queues = metrics.Gauge('indico_size_livesync_queues', 'Items in Livesync queues')
+
+
+def update_metrics(active_user_hours):
+    """Update all metrics."""
+    now = now_utc()
+    num_events.set(Event.query.filter(~Event.is_deleted).count())
+    num_active_events.set(Event.query.filter(~Event.is_deleted, Event.start_dt <= now, Event.end_dt >= now).count())
+    num_users.set(User.query.filter(~User.is_deleted).count())
+    num_active_users.set(
+        User.query
+        .filter(Identity.last_login_dt > (now - timedelta(hours=active_user_hours)))
+        .join(Identity).group_by(User).count()
+    )
+    num_categories.set(Category.query.filter(~Category.is_deleted).count())
+
+    num_active_attachment_files.set(get_attachment_query().count())
+    num_attachment_files.set(
+        AttachmentFile
+        .query
+        .join(Attachment, AttachmentFile.attachment_id == Attachment.id)
+        .count()
+    )
+
+    attachment_subq = db.aliased(Attachment, get_attachment_query().subquery('attachment'))
+
+    size_active_attachment_files.set(
+        db.session.query(db.func.sum(AttachmentFile.size))
+        .filter(AttachmentFile.id == attachment_subq.file_id)
+        .scalar() or 0
+    )
+
+    size_attachment_files.set(
+        db.session.query(db.func.sum(AttachmentFile.size))
+        .join(Attachment, AttachmentFile.attachment_id == Attachment.id)
+        .scalar() or 0
+    )
+
+    if LIVESYNC_AVAILABLE:
+        size_livesync_queues.set(LiveSyncQueueEntry.query.filter(~LiveSyncQueueEntry.processed).count())
+
+    num_notes.set(get_note_query().count())
+
+    num_rooms.set(Room.query.filter(~Room.is_deleted).count())
+    num_active_rooms.set(Room.query.filter(~Room.is_deleted, Room.is_reservable).count())
+
+    num_bookings.set(Reservation.query.filter(~Room.is_deleted).join(Room).count())
+    num_valid_bookings.set(Reservation.query.filter(~Room.is_deleted, ~Reservation.is_rejected).join(Room).count())
+    num_pending_bookings.set(
+        Reservation.query.filter(
+            ~Room.is_deleted,
+            Reservation.is_pending,
+            ~Reservation.is_archived
+        ).join(Room).count()
+    )
+
+    num_occurrences.set(ReservationOccurrence.query.count())
+    num_valid_occurrences.set(
+        ReservationOccurrence
+        .query
+        .filter(~Room.is_deleted, ~Reservation.is_rejected, ReservationOccurrence.is_valid)
```

Comment:
I think you also want to exclude cancelled ones here...  or filter by `Reservation.is_accepted` instead?

Reply:
But can a reservation be pending and cancelled at the same time? It's a single state field.

Reply:
Good point.